api/common/configentry_webhook.go

```diff
@@ -0,0 +1,53 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"k8s.io/api/admission/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// ConfigEntryLister is implemented by CRD-specific webhooks.
+type ConfigEntryLister interface {
+	// List returns all resources of this type across all namespaces.
+	List(ctx context.Context) ([]ConfigEntryResource, error)
+}
+
+// ValidateConfigEntry validates cfgEntry. It is a generic method that
+// can be used by all CRD-specific validators.
+// Callers should pass themselves as validator and kind should be the custom
+// resource name, e.g. "ServiceDefaults".
+func ValidateConfigEntry(
+	ctx context.Context,
+	req admission.Request,
+	logger logr.Logger,
+	validator ConfigEntryLister,
+	cfgEntry ConfigEntryResource,
+	kind string) admission.Response {
+
+	if req.Operation == v1beta1.Create {
```

Comment:
Added
```
	// On create we need to validate that there isn't already a resource with
	// the same name in a different namespace since we need to map all Kube
	// resources to a single Consul namespace.
```